src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs

```diff
@@ -771,5 +772,113 @@ public Lines(string? line1, string? line2)
                 ExpectedDiagnostic.Create(DereferencePossiblyNullReferenceSuppressor.SuppressionDescriptors["CS8600"]),
                 testCode);
         }
+
+        [TestCase("class", "CS8634")]
+        [TestCase("notnull", "CS8714")]
+        public void TestIssue462Suppressed(string constraint, string diagnostic)
+        {
+            var testCode = TestUtility.WrapMethodInClassNamespaceAndAddUsings($@"
+                [Test]
+                public void Test()
+                {{
+                    object? possibleNull = GetNext();
+                    object? assertedNotNull = possibleNull;
+                    Assert.NotNull(assertedNotNull);
+                    ↓DoNothing(assertedNotNull);
+                }}
+
+                private static object? GetNext() => default;
+
+                private static void DoNothing<T>(T s)
+                    where T : {constraint}
+                {{ }}
+            ");
+
+            RoslynAssert.Suppressed(suppressor,
+                ExpectedDiagnostic.Create(DereferencePossiblyNullReferenceSuppressor.SuppressionDescriptors[diagnostic]),
+                testCode);
+        }
+
+        [TestCase("class", "CS8634")]
+        [TestCase("notnull", "CS8714")]
+        public void TestIssue462AlsoSuppressed(string constraint, string diagnostic)
+        {
+            var testCode = TestUtility.WrapMethodInClassNamespaceAndAddUsings($@"
+                [Test]
+                public void Test()
+                {{
+                    object? possibleNull = GetNext();
+                    object? assertedNotNull = GetNext();
+                    Assert.NotNull(assertedNotNull);
+                    ↓DoNothing(assertedNotNull, possibleNull);
+                }}
+
+                private static object? GetNext() => default;
+
+                private static void DoNothing<T1, T2>(T1 p1, T2 p2)
+                    where T1 : {constraint}
+                    where T2 : class?
+                {{ }}
+            ");
+
+            RoslynAssert.Suppressed(suppressor,
+                ExpectedDiagnostic.Create(DereferencePossiblyNullReferenceSuppressor.SuppressionDescriptors[diagnostic]),
+                testCode);
+        }
+
+        [TestCase("class", "CS8634")]
+        [TestCase("notnull", "CS8714")]
+        public void TestIssue462SuppressesMultiple(string constraint, string diagnostic)
```

Comment:
I originally had a question about this in the production code, but this test answered it 👍